### PR TITLE
Prompt registry

### PR DESCRIPTION
**Pull request by hteeyeoh**
## Description

Prompt Registry microservice helps us to connect to a MongoDB Database and save the user most-used/favorite prompt and retrieve back from database thru keyword searching etc. This microservice can be used along with chat specific OPEA use cases for data persistence.

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

n/a

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
 [✓] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

motor==3.4.0

## Tests

Describe the tests that you ran to verify your changes.

Added test file for the microservice.

### CODE REVIEW

**Review by amberjain1 — APPROVED** [review]

File: comps/prompt_registry/mongo/prompt.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        user (str): The user of the requested prompt.
+        prompt_id (str): The prompt_id of prompt to be retrieved from database.
+    """
+
+    user: str
+    prompt_id: Optional[str] = None
+    prompt_text: Optional[str] = None
+
+
+@register_microservice(
```
Comment (lvliang-intel) [inline comps/prompt_registry/mongo/prompt.py:36]: We will support multiple endpoints within the same microservice. Starting a separate microservice for each endpoint is not practical.

File: comps/prompt_registry/mongo/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+curl -X 'POST' \
+  http://{host_ip}:6013/v1/prompt/get \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "user": "test"}'
+```
+
+```bash
+curl -X 'POST' \
```
Comment (lvliang-intel) [inline comps/prompt_registry/mongo/README.md:75]: The 'Get' operation should use a 'GET' request, not 'POST'.

Comment (hteeyeoh) [reply comps/prompt_registry/mongo/README.md:75]: Is GET supported in supported in register_microservice? As per I understand last time they are not supported and my prompt_registry design refer this design: [README](https://github.com/opea-project/GenAIComps/pull/294#discussion_r1685887980) as baseline.

File: comps/prompt_registry/mongo/prompt.py (hunk 1 above)
Comment (hteeyeoh) [reply comps/prompt_registry/mongo/prompt.py:36]: This current design is referenced from other components like dataprep design for now to keep aligned and avoid confusion. For multiple endpoint within same microservices, we can make appropriate changes when it is supported across. Is that ok?

File: tests/test_prompt_registry_mongo.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        echo "Correct result."
+    else
+        echo "Incorrect result."
+        exit 1
+    fi
+
+}
+
+function stop_docker() {
+    cid=$(docker ps -aq --filter "name=mongo*")
```
Comment (chensuyue) [inline tests/test_prompt_registry_mongo.sh:49]: Please also clean up `promptregistry-mongo-server`, so suggest to use `test-comps-` as container prefix, so here you can use ` cid=$(docker ps -aq --filter "name=test-comps-*")` to remove all the test images.

Comment (hteeyeoh) [reply tests/test_prompt_registry_mongo.sh]: Done changes.

**Review by ftian1 — COMMENTED** [review]
may i know how this service will be used from user view to construct an entire GenAI pipeline?

**Comment by hteeyeoh** [comment]
> may i know how this service will be used from user view to construct an entire GenAI pipeline?

Basically user can store their desired prompt which could help them to get better output into the registry and can retrieve the prompt back via keyword search without having to re-enter whenever they sending a query. Of course, this can be depends on UI design on how to integrate this for better user experience.

**Review by yogeshmpandey — APPROVED** [review]

**Review by ftian1 — APPROVED** [review]

File: comps/prompt_registry/mongo/README.md (hunk 2 above)
Comment (hteeyeoh) [reply comps/prompt_registry/mongo/README.md:75]: @lvliang-intel , are there still  any concern on this comment? Can you please check it.

File: comps/prompt_registry/mongo/prompt.py (hunk 1 above)
Comment (hteeyeoh) [reply comps/prompt_registry/mongo/prompt.py:36]: @lvliang-intel , are there still  any concern on this comment? Can you please check it.

**Pull request merged**